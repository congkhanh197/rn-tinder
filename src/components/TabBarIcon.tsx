import React from 'react';
import Icons from 'react-native-vector-icons/Ionicons';
import Colors from '../constants/Colors';

export interface IProps {
  name: string;
  focused: boolean;
}

const TabBarIcon = (props: IProps) => {
  return (
    <Icons
      name={props.name}
      size={26}
      color={props.focused ? Colors.tabIconSelected : Colors.tabIconDefault}
    />
  );
};

export default TabBarIcon;
