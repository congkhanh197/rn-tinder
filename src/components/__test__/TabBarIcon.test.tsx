import React from 'react';
import TabBarIcon, { IProps } from '../TabBarIcon';
import renderer from 'react-test-renderer';
describe('Test Tab bar icon', () => {
  test('should render with focused', () => {
    const props: IProps = {
        name: "testName",
        focused: true,
    }
    const tree = renderer.create(<TabBarIcon {...props}/>);
    expect(tree).toMatchSnapshot();
  });
  test('should render no focused', () => {
    const props: IProps = {
        name: "testName",
        focused: false,
    }
    const tree = renderer.create(<TabBarIcon {...props}/>);
    expect(tree).toMatchSnapshot();
  });
});
