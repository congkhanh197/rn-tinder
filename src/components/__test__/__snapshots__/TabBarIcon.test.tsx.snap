// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Tab bar icon should render no focused 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#ccc",
        "fontSize": 26,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  ?
</Text>
`;

exports[`Test Tab bar icon should render with focused 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#2f95dc",
        "fontSize": 26,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  ?
</Text>
`;
